tools/serverpod_cli/lib/src/analyzer/entities/validation/restrictions.dart

```diff
@@ -402,6 +461,39 @@ class Restrictions {
     return [];
   }
 
+  List<SourceSpanSeverityException> validateViewQuery(
+    String parentNodeName,
+    dynamic query,
+    SourceSpan? span,
+  ) {
+    if (query is! String) {
+      return [SourceSpanSeverityException('The "query" must be String.', span)];
+    }
+
+    if (query.isEmpty) {
+      return [
+        SourceSpanSeverityException(
+          'The "query" cannot be empty.',
+          span,
+        )
+      ];
+    }
+
+    // Check basic query if contains "SELECT" and "FROM" clauses
+    var selectPattern = RegExp(r'\bselect\b', caseSensitive: false);
+    var fromPattern = RegExp(r'\bfrom\b', caseSensitive: false);
```

Comment:
Question: The column names used in the query need to match the fields of the model for our deserialization to work correctly. Could we add validation for that?

Reply:
Yes, let me add validation to match the field with the query that the user defined in Yaml.